### ISSUE TITLE
Update.sh ==> Error 

### ISSUE THREAD

**Issue by Anima79** [mailcow/mailcow-dockerized#1107]
Hello,
I installed a fresh copy of Mailcow Dockerized yesterday. After the dev merge to master I tried to run the update.sh today - but it throws the following error at me: 

```
Committing current status... Fetching updated code from remote... From https://github.com/mailcow/mailcow-dockerized
 * branch            master     -> FETCH_HEAD
Merging local with remote code (recursive, strategy: "theirs", options: "patience"...
error: Your local changes to the following files would be overwritten by merge:
        data/conf/rspamd/override.d/worker-controller-password.inc
Please commit your changes or stash them before you merge.
Aborting

Oh no, what happened?
=> You most likely added files to your local mailcow instance that were now added to the official mailcow repository. Please move them to another location before updating mailcow.
```

What could be the cause of this? Should I delete the file?

**Comment by Anima79** [comment]
Just deleting the file helped in my case -- I've been able to update my system now. Afterwards setting up a new password for rspamd had to be done.

**Comment by MAGICCC** [comment]
You could also use `git add` / `git commit` and such

**Comment by andryyy** [comment]
The update script should ignore that file. Can you try to run the update again and check what happens?

**Comment by Anima79** [comment]
I ran the script now over an already updated system - no problems anymore. Weird! However, there is still one problem: in the rspamd UI, when trying to open the "Throughput"-window, there comes the following message:

`Cannot receive RRD data from: local, error: No rrd configured for graphs`

**Comment by andryyy** [comment]
Is there a `rspamd-vol-1:/var/lib/rspamd` in your `docker-compose.yml` under `rspamd-mailcow`?

**Comment by Anima79** [comment]
Yes.

**Comment by andryyy** [comment]
And what's in your logs? `docker-compose logs --tail=400 rspamd-mailcow`  after you open throughput.

**Comment by Anima79** [comment]
Hi andryyy,
after restarting and repulling the image it works now. The logs seem to be quite normal - except of the following entry:

`2018-02-28 20:31:28 #12(controller) <ec2301>; csession; rspamd_controller_check_password: absent password has been specified` 

Is this related to me deleting the data/conf/rspamd/override.d/worker-controller-password.inc-file?

Sorry for stealing your time!! ... and many thanks in advance!

**Comment by Anima79** [comment]
Nope, the error came again: 

```
rspamd-mailcow_1     | 2018-02-28 20:28:29 #14(normal) <815be1>; task; rspamd_protocol_http_reply: regexp statistics: 0 pcre regexps scanned, 5 regexps matched, 164 regexps total, 80 regexps cached, 0B bytes scanned using pcre, 5.45k bytes scanned total
rspamd-mailcow_1     | 2018-02-28 20:28:31 #11(rspamd_proxy) <63b0c9>; proxy; proxy_milter_finish_handler: finished milter connection
rspamd-mailcow_1     | 2018-02-28 20:31:28 #12(controller) <ec2301>; csession; rspamd_controller_check_password: absent password has been specified
rspamd-mailcow_1     | 2018-02-28 20:35:00 #12(controller) <oxhd99>; map; http_map_finish: http://nginx:8081/settings.php(172.22.1.10:8081): read map data 256 bytes, next check at Wed, 28 Feb 2018 20:35:30 GMT
rspamd-mailcow_1     | 2018-02-28 20:35:00 #12(controller) <a1gq9u>; lua; settings.lua:610: loaded 1 elements of settings
rspamd-mailcow_1     | 2018-02-28 20:35:01 #15(normal) <oxhd99>; map; rspamd_map_read_cached: http://nginx:8081/settings.php: read map data cached 256 bytes
rspamd-mailcow_1     | 2018-02-28 20:35:01 #15(normal) <a1gq9u>; lua; settings.lua:610: loaded 1 elements of settings
rspamd-mailcow_1     | 2018-02-28 20:35:01 #11(rspamd_proxy) <oxhd99>; map; rspamd_map_read_cached: http://nginx:8081/settings.php: read map data cached 256 bytes
rspamd-mailcow_1     | 2018-02-28 20:35:01 #11(rspamd_proxy) <a1gq9u>; lua; settings.lua:610: loaded 1 elements of settings
rspamd-mailcow_1     | 2018-02-28 20:35:01 #13(controller) <eef871>; csession; rspamd_controller_handle_graph: no rrd configured
rspamd-mailcow_1     | 2018-02-28 20:35:01 #13(controller) logger; file_log_function: Last message repeated 12 times
rspamd-mailcow_1     | 2018-02-28 20:35:01 #13(controller) <oxhd99>; map; rspamd_map_read_cached: http://nginx:8081/settings.php: read map data cached 256 bytes
rspamd-mailcow_1     | 2018-02-28 20:35:01 #13(controller) <a1gq9u>; lua; settings.lua:610: loaded 1 elements of settings
rspamd-mailcow_1     | 2018-02-28 20:35:05 #14(normal) <oxhd99>; map; rspamd_map_read_cached: http://nginx:8081/settings.php: read map data cached 256 bytes
rspamd-mailcow_1     | 2018-02-28 20:35:05 #14(normal) <a1gq9u>; lua; settings.lua:610: loaded 1 elements of settings
rspamd-mailcow_1     | 2018-02-28 20:35:34 #12(controller) <oxhd99>; map; rspamd_map_read_cached: http://nginx:8081/settings.php: read map data cached 256 bytes
rspamd-mailcow_1     | 2018-02-28 20:35:34 #12(controller) <a1gq9u>; lua; settings.lua:610: loaded 1 elements of settings
rspamd-mailcow_1     | 2018-02-28 20:35:49 #13(controller) <d3fbd6>; csession; rspamd_controller_handle_graph: no rrd configured
rspamd-mailcow_1     | 2018-02-28 20:35:53 #13(controller) <5728df>; csession; rspamd_controller_handle_graph: no rrd configured
```

**Comment by andryyy** [comment]
Does this file exists now? You should not delete it... touch it as empty file but don't delete it. It probably was created as folder now. Check if that former file is now a folder and delete it. Run `touch data/conf/rspamd/override.d/worker-controller-password.inc` afterwards and restart Rspamd.

Now please run `docker-compose exec rspamd-mailcow ls -li /var/lib/rspamd` and show me the output.

Run `docker-compose exec rspamd-mailcow bash -c "chown -R _rspamd:_rspamd /var/lib/rspamd/"` to fix it and restart rspamd-mailcow.

**Comment by Anima79** [comment]
What a strange error. After a delay of half a minute or so the graph shows itself. But when I choose for example "local", then the error shows up again. 

P.S.: Saw your answer just a moment ago. Will follow your advice now,

**Comment by Anima79** [comment]
Hmm, tried your steps, but no luck. I even can't login into the rspamd UI right now, although I changed the password. 

Here is the result of `docker-compose exec rspamd-mailcow ls -li /var/lib/rspamd`
```

19661702 -rw-r--r-- 1 _rspamd _rspamd  797696 Feb 28 20:54 rspamd.rrd
19661698 srw------- 1 root    root          0 Feb 28 20:54 rspamd.sock
19661992 -rw-r--r-- 1 _rspamd _rspamd     172 Feb 28 20:54 stats.ucl
19661696 -rw-r--r-- 1 _rspamd _rspamd   48912 Feb 28 20:54 symbols.cache
```
Chowning doesn't change the ownership of root.

**Comment by Anima79** [comment]
Okay, I restored the original data/conf/rspamd/override.d/worker-controller-password.inc from a backup I made before deleting the file. I can now log into the rspamd UI again. 

However, the error with `Cannot receive RRD data from: local, error: No rrd configured for graphs` sometimes (!) showing up persists. Since this is not a real problem per se, I will hope that some future update will get rid of it.
